modules/Bio/EnsEMBL/DBSQL/TranscriptSupportingFeatureAdaptor.pm

```diff
@@ -99,23 +104,34 @@ sub fetch_all_by_Transcript {
   while(my ($type, $feature_id) = $sth->fetchrow){
     if($type eq 'protein_align_feature'){
       $feature = $prot_adp->fetch_by_dbID($feature_id);
-    } elsif($type eq 'dna_align_feature'){
+     } elsif($type eq 'dna_align_feature'){
       $feature = $dna_adp->fetch_by_dbID($feature_id);
     } else {
       warning("Unknown feature type [$type]\n");
     }
-
+    
     if(!$feature) {
       warning("Supporting feature $type $feature_id does not exist in DB");
     } else {
       my $new_feature = $feature->transfer($transcript->slice());
-      push @$out, $new_feature if( $new_feature );
+ 
+      push @{$out_feature_type->{$type}}, $new_feature if ($new_feature);
     }
   }
-
-  $sth->finish();
-
-  return $out;
+	  $sth->finish();
+
+   
+   if(defined $feature_type){
+   	return $out_feature_type->{$feature_type};
+   }else{
+   
+  	while(my ($feature_type, $new_features) = each(%$out_feature_type)){
+  		push @$out, @{$new_features};
```

Comment:
We have a no \t policy in this repo. It looks like this is tabbed in. Preferred substitution is 4 or 2 spaces.